src/autopas/containers/linkedCells/traversals/LCC08CellHandlerUtility.h

```diff
@@ -0,0 +1,150 @@
+enum class C08OffsetMode {
+  /** Returns the C08 base step cell pairs without sorting */
+  C08_CELL_PAIRS = 0,
+  /** Returns the C08 base step cell pairs with sorting (for SortedView projection) */
+  C08_CELL_PAIRS_SORTING = 1,
+  /** Returns the C08 base step cell pairs adapted to C04, i.e. two-dimensiona resolved on X-axis */
+  C04_CELL_PAIRS = 2,
+};
+
+/**
+ * Template Magic Parameter Alias which links the types {@link OffsetPairSorting}, {@link OffsetPair} and {@link
+ * OffsetPairVector}
+ */
+template <C08OffsetMode Mode>
+using OffsetPairType = std::vector<
+    std::conditional_t<Mode == C08OffsetMode::C08_CELL_PAIRS_SORTING, OffsetPairSorting,
+                       std::conditional_t<Mode == C08OffsetMode::C04_CELL_PAIRS, OffsetPairVector, OffsetPair>>>;
+
+/**
+ * Represents the interaction directions between cell pairs in the C08 base step
+ */
+enum class C08CellDirection : int {
+  /** The origin or the base cell in the four-cell-square **/
+  FRONT_LEFT,
+  /** The cell to the back from the origin **/
+  BACK_LEFT,
+  /** The cell to the right from the origin **/
+  FRONT_RIGHT,
+  /** The cell diagonally to the origin **/
+  BACK_RIGHT,
+};
+
+/**
+ * Array containing all four enum values of {@link autopas::internal::C08CellDirection}.
+ */
+constexpr inline std::array<C08CellDirection, 4> ALL_DIRECTIONS{
+    {C08CellDirection::FRONT_LEFT, C08CellDirection::BACK_LEFT, C08CellDirection::FRONT_RIGHT,
+     C08CellDirection::BACK_RIGHT}};
+
+/**
+ * Calculates the offset for cell2 in the pair, given a direction from the base cell, a z index and the overlap.
+ * @param overlap1 the overlap of the interacting cells (plus one)
+ * @param direction one of the four directions
+ * @param z the z index of the spatial dimension
+ * @return offset of cell2
+ */
+constexpr int calculateCell2Index(int overlap1, const C08CellDirection &direction, int z);
+
+/**
+ * Calculates the multipliers (zero or one) for a given direction.
+ * This is used as a helper to calculate the `distVec` in {@link autopas::internal::computePairwiseCellOffsetsC08}.
+ * @param direction one of the four directions
+ * @return pair of multipliers: first one for x dimension, second one for y dimension
+ */
+constexpr std::pair<int, int> toDistVectorMultiplier(const C08CellDirection &direction);
+
+/**
+ * Returns true if the cell-interaction in the given direction, with the given base cell coordinates and overlap
+ * needs to be included into the pairwise cell offsets
+ * @param direction the direction from base cell to inetracting cell
+ * @param overlap the overlap (calculated from interactionLength divided by cellLength)
+ * @param x the x offset of the base cell
+ * @param y the y offset of the base cell
+ * @param z the z offset of the base cell
+ * @return true if the pair needs to be included
+ */
+constexpr bool includeC08Case(const C08CellDirection &direction, const std::array<int, 3> &overlap, int x, int y,
```

Comment:
The doc is a bit too general and doesn't tell me more than the function name itself. What is the decision based on? Or sharpen the name.